posthog/clickhouse/client/execute.py

```diff
@@ -108,12 +110,21 @@ def sync_execute(
             )
         except Exception as err:
             err = wrap_query_error(err)
-            statsd.incr(
-                "clickhouse_sync_execution_failure",
-                tags={"failed": True, "reason": type(err).__name__},
-            )
-
-            raise err
+            if type(err).__name__ == "CHQueryErrorQueryWasCancelled":
+                statsd.incr("clickhouse_execution_cancelled", tags={"team_id": team_id})
+                raise err
+            elif type(err).__name__ == "EstimatedQueryExecutionTimeTooLong":
+                statsd.incr(
+                    "clickhouse_execution_timeout",
+                    tags={"team_id": team_id},
+                )
+                raise err
+            else:
+                statsd.incr(
+                    "clickhouse_sync_execution_failure",
+                    tags={"failed": True, "reason": type(err).__name__},
+                )
+                raise err
         finally:
```

Comment:
I've just been on a spree this week looking into issues causing this errors, and there were some very real and painful root causes. Some are fixed already and I have fixes some mor. Sentry helped there, so I think we should revert this part of the diff and keep raising these as we did